timemachine/fe/model_rabfe.py

```diff
@@ -580,5 +583,290 @@ def setup_topology(self, mol):
 
 class RelativeBindingModel(RelativeModel):
     def setup_topology(self, mol_a, mol_b):
-        top = topology.DualTopologyStandardDecoupling(mol_a, mol_b, self.ff)
+        top = topology.DualTopologyDecoupling(mol_a, mol_b, self.ff)
         return top
+
+
+class RelativeConversionModel:
+    """
+    Estimate the free energy of decharging molecule A and charging molecule B
+    while both molecules are in the binding pocket.
+    """
+
+    def __init__(
+        self,
+        client: Optional[AbstractClient],
+        ff: Forcefield,
+        host_system: openmm.System,
+        host_schedule: np.ndarray,
+        host_topology: openmm.app.Topology,
+        temperature: float,
+        pressure: float,
+        dt: float,
+        equil_steps: int,
+        prod_steps: int,
+        frame_filter: Optional[callable] = None,
+        k_core: float = 30.0,
+    ):
+
+        self.host_system = host_system
+        self.temperature = temperature
+        self.pressure = pressure
+        self.dt = dt
+        self.host_schedule = host_schedule
+        self.host_topology = host_topology
+        self.client = client
+        self.ff = ff
+        self.equil_steps = equil_steps
+        self.prod_steps = prod_steps
+        if frame_filter is None:
+            frame_filter = endpoint_frames_only
+        self.frame_filter = frame_filter
+        self.k_core = k_core
+
+    def setup_topology(self, mol_a, mol_b):
+        top = topology.DualTopologyChargeConversion(mol_a, mol_b, self.ff)
+        return top
+
+    def _futures_a_to_b(self, ff_params, mol_a, mol_b, combined_core_idxs, x0, box0, prefix, seed):
```

Comment:
Can this be inherited rather than duplicated?

Reply:
yes but I will do this in a separate class and take care of the other hydration classes as well hopefully!